src/distPlus/distribution/AlgebraicShapeCombination.re

```diff
@@ -219,35 +226,21 @@ let toDiscretePointMassesFromDiscrete = (s: DistTypes.xyShape): pointMassesWithM
   let variances: array(float) = Belt.Array.makeUninitializedUnsafe(n);
 
   for (i in 0 to n - 1) {
-    let _ =
-      Belt.Array.set(
-        masses,
-        i,
-        ys[i]
-      );
-
-    let _ =
-      Belt.Array.set(
-        means,
-        i,
-        xs[i]
-      );
-
-    let _ =
-      Belt.Array.set(
-        variances,
-        i,
-        0.0
-      );
-    ();
+    Belt.Array.set(masses, i, ys[i]) |> ignore;
+    Belt.Array.set(means, i, xs[i]) |> ignore;
+    Belt.Array.set(variances, i, 0.0) |> ignore;
   };
 
   {n, masses, means, variances};
 };
 
-let combineShapesContinuousDiscreteAdd =
-    (op: ExpressionTypes.algebraicOperation, s1: DistTypes.xyShape, s2: DistTypes.xyShape)
-  : DistTypes.xyShape => {
+let combineShapesContinuousDiscrete =
+    (
+      op: ExpressionTypes.algebraicOperation,
+      s1: DistTypes.xyShape,
+      s2: DistTypes.xyShape,
+    )
+    : array(DistTypes.xyShape) => {
```

Comment:
I don't really understand this function. It seems to return an array of shapes, which requires additional external manipulation outside this function to return a meaningful result? This seems confusing to me.

Reply:
It's super confusing, and part of the whole problem with the discrete/continuous case. The issue is that the array of returned shapes needs to be pointwise-combined (which happens in [Continuous.re:264](https://github.com/foretold-app/widedomain/blob/1dfea41ab89b9a757e7ff454354dd58e2bd6b05b/src/distPlus/distribution/Continuous.re#L264)), but the interpolation strategy (linear vs. pointwise) depends on the `continuousShape`. But `combineShapesContinuousDiscrete` isn't aware of the interpolation stuff because it operates only on the xyShapes. 

If we want this to be clean and self-explanatory then we really need to rethink the way these functions are broken up.

Reply:
Can we just use sampling for all of this for now? I'm fine doing sampling in all cases where there's more than one point in the discrete distribution (in which case we can make a few simplifying assumptions). 

I imagine this form only makes sense for discrete distributions with between 2 to 40 points or similar, which seems  unusual.